frappe/public/js/frappe/upload.js

```diff
@@ -296,6 +296,16 @@ frappe.upload = {
 		if (opts.no_socketio || frappe.flags.no_socketio || file_not_big_enough) {
 			upload_with_filedata();
 			return;
+		} else {
+			args.file_size = fileobj.size;
+			frappe.call({
+				method: 'frappe.utils.file_manager.validate_filename',
+				async: false,
+				args: {"filename": args.filename},
+				callback: function(r) {
+					args.filename = r.message;
+				}
+			});
 		}
 
 		frappe.socketio.uploader.start({
```

Comment:
upload should only start after filename validation, otherwise `args.filename` will not change.